// inventory_client.proto (para comunicarse con el servicio de inventario)
syntax = "proto3";

package inventory;

service InventoryService {
  rpc GetBook (BookRequest) returns (BookResponse) {}
  rpc UpdateStock (UpdateStockRequest) returns (UpdateStockResponse) {}
}

message BookRequest {
  string book_id = 1;
}

message BookResponse {
  string book_id = 1;
  string title = 2;
  string author = 3;
  string isbn = 4;
  int32 stock = 5;
  double price = 6;
}

message UpdateStockRequest {
  string book_id = 1;
  int32 quantity_change = 2;
}

message UpdateStockResponse {
  bool success = 1;
  int32 new_stock = 2;
  string message = 3;
}
